### DISCUSSION TITLE
Upcoming Birt-Release planned working with Tomcat 10?

### DISCUSSION THREAD

**Discussion by mr0reed137** [eclipse-birt/birt#1445] [Q&A]
Hello,

I'm curious to know what's next for Birt, especially if there's already a next version planned that includes functionality with Tomcat 10 (and of course java > 17).

Thanks for any feedback on this.

Marco

**Comment by speckyspooky** [comment]
The upcoming version BIRT 4.14 will work with Tomcat 8.x and 9.x but not with Tomcat 10.
The reason of that that the Tomcat 10 based in full of the new jarkarta-EE-project and for that BIRT must be migrated completely.

**Reply by mr0reed137** [reply]
Thanks @speckyspooky for the reply. 
I know about Birt 4.14 and looking forward to that version. 
But are you planning on releasing a version 4.15 or greater, which will work with tomcat 10, anytime soon? Any plans on that?

**Reply by speckyspooky** [reply]
There isn't a time defined plan of BIRT 4.15. Yes, we talked about in some discussions but there are bigger challenges.
One of the main topics is that lot of BIRT developers use java without jarkarta and also the eclipse project which is the base for the editor isn't full migrated to jarkarta.
I will say, yes the topic is in scope but it will needs time to migrate all the components to jarakarta and to be able to run with Tomcat 10.

**Reply by merks** [reply]
Note that the platform has fully migrated to Jetty 12 for 4.30/2023-12 so I think will be possible soon to move forward on an issue like this.  Though I'm really not sure what is actually required to run with Tomcat 10. 

I'm currently working on easing the migration to Jetty 12 which includes the option to use ee8, ee9, or ee10:

https://github.com/eclipse-orbit/orbit-simrel/issues/19

**Reply by claesrosell** [reply]
The following is my personal thoughts around this. 
This is unfortunately much bigger then just the switch of namespace since we are relying on legacy dependencies which will not switch to Jakarta.*. Like Axis1.
Since BIRT depends on Axis1 and for it's SOAP based web services and also uses JSP for it's HTML rendering the whole web stack is dated. For BIRT to stay relevant we would probably need to make some big architectural changes like REST based services and more JavaScript focused rendering. This would break backwards compatibility.  The team would also need to agree on the technical solutions. In other words this is a huge task.

**Reply by speckyspooky** [reply]
Definitley is the change to JarkartaEE a hugh change and I agree with @claesrosell this would be a new "way" of BIRT with according consequences.
I don't know if we must directly switch the whole web architecture. My preveration would be to do it into two different steps. But of course this part of BIRT should be get a full review with the switch to JarkartaEE.
